### ISSUE TITLE
[ts-react-apollo] Types should able to import from other file when it is the only plugin used

### ISSUE THREAD

**Issue by veeramarni** [dotansimha/graphql-code-generator#2165]
**Is your feature request related to a problem? Please describe.**

If we need to generate an independent file with only `typescript-react-apollo` plugin, it needs types generated from other plugins to be imported externally. 

**Describe the solution you'd like**

For example, if we have an option like `typesImport: @external-file` in `typescript-react-apollo` plugin

```codegen.yml
generates:
  packages/sample-client-state/src/interfaces/generated-models.ts:
    documents: "packages/sample-client-state/src/**/*.gql"
    schema: 
      - ./packages/sample-client-state/src/**/*.graphql
    config:
      enumValues: ./enum
      constEnums: true
      skipTypename: true
      noNamespaces: true
      contextType: ./context#MyContext
      typesPrefix: I
      withMutationFn: false
      withHOC: false
      withComponent: false
    plugins:
      - add: |
          /* tslint:disable */
      - typescript
      - typescript-operations
      - typescript-resolvers
      - typescript-react-apollo
  packages/sample-client-react/src/generated/models.tsx:
    documents: "packages/sample-client-state/src/**/*.gql"
    schema: 
      - ./packages/sample-client-state/src/**/*.graphql
    config:
        typesPrefix: I
        typesImport: @external-file
        withMutationFn: true
        withHOC: true
        withComponent: true
        noGraphQLTag: true
    plugins:
      - typescript-react-apollo
```

generated `.ts` file
```
.....
export type ISelectFileMutationVariables = {
  content?: Maybe<ISelectEditor_Input>;
};

export type ISelectFileMutation = Pick<IMutation, "selectFile">;
....
```

generated `.tsx` file
```
import { ISelectFileMutation, ISelectFileMutationVariables } from '@external-file'
....
export type ISelectFileMutationFn = ReactApollo.MutationFn<
  ISelectFileMutation,
  ISelectFileMutationVariables
>;
export type SelectFileComponentProps = Omit<
  ReactApollo.MutationProps<ISelectFileMutation, ISelectFileMutationVariables>,
  "mutation"
>;

export const SelectFileComponent = (props: SelectFileComponentProps) => (
  <ReactApollo.Mutation<ISelectFileMutation, ISelectFileMutationVariables>
    mutation={SelectFileDocument}
    {...props}
  />
);

export type ISelectFileProps<TChildProps = {}> = Partial<
  ReactApollo.MutateProps<ISelectFileMutation, ISelectFileMutationVariables>
> &
  TChildProps;
export function withSelectFile<TProps, TChildProps = {}>(
  operationOptions?: ReactApollo.OperationOption<
    TProps,
    ISelectFileMutation,
    ISelectFileMutationVariables,
    ISelectFileProps<TChildProps>
  >
) {
  return ReactApollo.withMutation<
    TProps,
    ISelectFileMutation,
    ISelectFileMutationVariables,
    ISelectFileProps<TChildProps>
  >(SelectFileDocument, {
    alias: "withSelectFile",
    ...operationOptions
  });
}
......
```
That way we have a `non-react` package that could be used for various purpose and for `react` component, we can generate a separate `.tsx` file by importing any types that required to generate `.tsx` through importing externally.

**Describe alternatives you've considered**

The alternative would be using `add` plugin and add the types manually which would be a cumbersome task to update the types every time types are changed or updated.
**Additional context**

**Comment by namesphill** [comment]
for me at least, it would work if there were a way to just not include the "export" keyword in the typescript generated files.
Here goes my `codegen.yml`:
```yml
schema: src/typeDefs.ts
documents: src/documents.ts
generates:
  ../client/@types/types.d.ts:
    plugins:
      - typescript
  ../client/codegen/operations.ts:
    plugins:
      - typescript-operations
config:
  typesPrefix: GQL_
  enumsAsTypes: true
  declarationKind: "interface"
  maybeValue: T | null | undefined
```
In my `.d.ts` all interfaces have the export keyword, which makes the whole file's types not be globally available throughout the project.
I could have a really great level of separation of concerns if I could have all my plugin's generate their own stuff in different files.
My proposal:
```yml
config:
   exportTypes: false
```

**Comment by dotansimha** [comment]
@namesphill can you please elaborate on removing `export`? I'm not sure how is that helpful - you don't want to export types? how will you access it? (related: https://github.com/dotansimha/graphql-code-generator/issues/2246)

@veeramarni you are right. We have an internal configuration flag for letting `typescript-operations` import types from `typescript` (called `namespacedImportName`, we are using it for `near-operation-file` preset).
I think adding a similar approach for the client-side wrappers might do the trick. PRs are welcome 😄

**Comment by namesphill** [comment]
Yeah. When you don't export and import anything, all your declarations in a given d.ts file become globally accessible. You don't need to import them to reference them. (Depending on your ts config) and that is something I really likes I think it would be great to have that as a config for the codegen.

**Comment by dotansimha** [comment]
@namesphill I added this feature for avoiding `export`, but note that we can't apply it to `typescript-react-apollo` plugin, because it creates variables, and they are not allowed in `.d.ts` files.

**Comment by veeramarni** [comment]
@dotansimha Can you please check this PR, it works fine for me.

**Comment by dotansimha** [comment]
Thanks @veeramarni , I'll review it soon :) sorry for the delay.

**Comment by veeramarni** [comment]
Ok

**Comment by dotansimha** [comment]
Available in 1.7.0

**Comment by veeramarni** [comment]
It seems the versions in the `package.json` pointing to `1.5.0` https://github.com/dotansimha/graphql-code-generator/blob/master/packages/presets/import-types/package.json#L12

I will send another PR to fix it.
